### PR TITLE
Improve API compatibility and client library reference docs

### PR DESCRIPTION
**Pull request by jstirnaman**
- Move API primers to API compatibility
- Update and remove duplication, link to examples
- Add Serverless v1 and retention policy usage
- Format placeholders
- Add Go v3 client library, Python v3 client library reference
- Cleanup JS v2 client library

### CODE REVIEW

**Comment by jstirnaman** [comment]
@sanderson Please check me on the retention policy use for Serverless. Whenever approved, merge it in when Serverless InfluxQL goes live.

File: content/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/browser.md [hunk 1]
```diff
@@ -19,30 +18,35 @@ related:
...
 - [Grafana](/influxdb/cloud-dedicated/query-data/tools/grafana/)
 - [InfluxQL with InfluxDB v1 HTTP API](/influxdb/cloud-dedicated/primers/api/v1/#query-using-the-v1-api)
 - [Chronograf](/{{< latest "Chronograf" >}}/)
 
 {{% /note %}}
 
+This library supports both front-end and server-side environments and provides the following distributions:
+* ECMAScript modules (ESM) and CommonJS modules (CJS)
+* Bundled ESM
+* Bundled UMD
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/browser.md:44]:
```suggestion
This library supports both front-end and server-side environments and provides the following distributions:

- ECMAScript modules (ESM) and CommonJS modules (CJS)
- Bundled ESM
- Bundled UMD 
```

File: content/influxdb/cloud-dedicated/reference/client-libraries/v3/_index.md [hunk 2]
```diff
@@ -13,11 +13,20 @@ influxdb/cloud-dedicated/tags: [client libraries, API, developer tools]
...
 
-Functionality varies among client libraries.
-These client libraries are in active development and may not be feature-complete.
+InfluxDB client libraries provide configurable batch writing of data to InfluxDB HTTP APIs.
+They can be used to construct line protocol data and transform data from other formats
+to line protocol.
+
+InfluxDB v3 client libraries can query InfluxDB v3 using the Arrow Flight protocol with gRPC.
+Client libraries use Flight clients to execute SQL and InfluxQL queries, request
+bucket information, and retrieve data stored in {{% cloud-name %}}.
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/reference/client-libraries/v3/_index.md:25]:
```suggestion
database information, and retrieve data stored in {{% cloud-name %}}.
```

File: content/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/nodejs/install.md [hunk 3]
```diff
@@ -49,22 +75,23 @@ Many of the client library examples use [TypeScript](https://www.typescriptlang.
...
+The `@influxdata/influxdb-client-apis` client library package won't work with {{% cloud-name %}}.
+It only works with InfluxDB v2 management APIs.
 
 ## Configure credentials
 
 The client examples include an [`env`](https://github.com/influxdata/influxdb-client-js/blob/master/examples/env.js) module for accessing your InfluxDB properties from environment variables or from `env.js`.
 The examples use these properties to interact with the InfluxDB API.
 
-Set environment variables or update `env.js` with your InfluxDB [database](/influxdb/cloud-dedicated/admin/databases/) (bucket), organization (required, but ignored), [token](/influxdb/cloud-dedicated/admin/tokens/), and cluster URL.
+Set environment variables or update `env.js` with your InfluxDB [database](/influxdb/cloud-serverless/admin/databases/) (bucket), organization (required, but ignored), [token](/influxdb/cloud-serverless/admin/tokens/), and cluster URL.
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/nodejs/install.md:94]:
```suggestion
Set environment variables or update `env.js` with your InfluxDB [database](/influxdb/cloud-dedicated/admin/databases/), organization (required, but ignored), [token](/influxdb/cloud-dedicated/admin/tokens/), and cluster URL.
```

File: content/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/nodejs/install.md [hunk 4]
```diff
@@ -79,6 +106,6 @@ Set environment variables or update `env.js` with your InfluxDB [database](/infl
 
 ## Next steps
 
-Once you've installed the client library and configured credentials, you're ready to [write data](/influxdb/cloud-dedicated/api-guide/client-libraries/nodejs/write/) to InfluxDB.
+Once you've installed the client library and configured credentials, you're ready to [write data](/influxdb/cloud-serverless/api-guide/client-libraries/nodejs/write/) to InfluxDB.
 
-{{< page-nav next="/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/nodejs/write/" keepTab=true >}}
+{{< page-nav next="/influxdb/cloud-serverless/reference/client-libraries/v2/javascript/nodejs/write/" keepTab=true >}}
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/nodejs/install.md:111]:
```suggestion
{{< page-nav next="/influxdb/cloud-dedicated/reference/client-libraries/v2/javascript/nodejs/write/" keepTab=true >}}
```

File: content/influxdb/cloud-dedicated/reference/client-libraries/v3/python.md [hunk 5]
```diff
@@ -7,18 +7,490 @@ menu:
...
+{{% code-placeholders "DATABASE_(NAME|TOKEN)" %}}
+```py
+with InfluxDBClient3(token="DATABASE_TOKEN", host="cluster-id.influxdb.io",
+                    org="", database="DATABASE_NAME",
+                    _write_client_options=wco) as client:
+
+    client.write(record=points)
+```
+{{% /code-placeholders %}}
+    {{< /tabs-wrapper >}}
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/reference/client-libraries/v3/python.md:140]: No tabs here.
~~~suggestion

    {{% code-placeholders "DATABASE_(NAME|TOKEN)" %}}
```py
with InfluxDBClient3(token="DATABASE_TOKEN", host="cluster-id.influxdb.io",
                    org="", database="DATABASE_NAME",
                    _write_client_options=wco) as client:

    client.write(record=points)
```
    {{% /code-placeholders %}}

~~~

File: content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md [hunk 6]
```diff
@@ -8,17 +8,490 @@ menu:
...
+{{% code-placeholders "BUCKET_(NAME|TOKEN)|API_TOKEN" %}}
+```py
+with  InfluxDBClient3(token="API_TOKEN", host="cloud2.influxdata.com",
+                      org="", database="BUCKET_NAME",
+                      _write_client_options=wco) as client:
+
+      client.write(record=points)
+```
+{{% /code-placeholders %}}
+    {{< /tabs-wrapper >}}
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md:140]:
~~~suggestion
    {{% code-placeholders "BUCKET_(NAME|TOKEN)|API_TOKEN" %}}
```py
with  InfluxDBClient3(token="API_TOKEN", host="cloud2.influxdata.com",
                      org="", database="BUCKET_NAME",
                      _write_client_options=wco) as client:

      client.write(record=points)
```
    {{% /code-placeholders %}}
~~~

File: content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md [hunk 7]
```diff
@@ -8,17 +8,490 @@ menu:
...
+##### Initialize a client
+
+The following example initializes a client for writing and querying the bucket.
+Given `_write_client_options=None`, the client will use synchronous mode when writing data.
+
+{{% code-placeholders "BUCKET_(NAME|TOKEN)|API_TOKEN" %}}
+```py
+client = InfluxDBClient3(token="API_TOKEN",
+                         host="cloud2.influxdata.com",
+                         org="",
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md:160]: Do they need their org name with serverless or is it inferred from the token?

File: content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md [hunk 8]
```diff
@@ -8,17 +8,490 @@ menu:
...
+    The file name must end with one of the supported extensions.
+    For more information about encoding and formatting data, see the documentation for each supported format.
+  
+    | Supported format                                                           | File name extension |
+    |:---------------------------------------------------------------------------|:--------------------|
+    | [Feather](https://arrow.apache.org/docs/python/feather.html)               | `.feather`          |
+    | [Parquet](https://arrow.apache.org/docs/python/parquet.html)               | `.parquet`          |
+    | [Comma-separated values](https://arrow.apache.org/docs/python/csv.html)    | `.csv`              |
+    | [JSON](https://pandas.pydata.org/docs/reference/api/pandas.read_json.html) | `.json`             |
+    | [ORC](https://arrow.apache.org/docs/python/orc.html)                       | `.orc`              |
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md:335]:
```suggestion
    | [ORC](https://arrow.apache.org/docs/python/orc.html)                       | `.orc`              |  
    
```

**Review by sanderson — APPROVED** [review]
Wow, this is awesome! Amazing work. Some minor suggestions here and there, but really great stuff.

File: content/influxdb/cloud-dedicated/reference/client-libraries/v3/python.md (hunk 5 above)
Comment (jstirnaman) [reply content/influxdb/cloud-dedicated/reference/client-libraries/v3/python.md:140]: > No tabs here.

I did this b/c `code-placeholders` breaks when indented. I thought `tabs-wrapper` was a workaround at the time, but I'm not sure about that now.

File: content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md (hunk 7 above)
Comment (jstirnaman) [reply content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md:160]: > Do they need their org name with serverless or is it inferred from the token?

Inferred from the token. Just tested again and it works when `org` is omitted.

```py
>>> client = InfluxDBClient3(host=host, org="", token=token, database=database)
>>> client.query('show tables')
pyarrow.Table
table_catalog: string not null
table_schema: string not null
table_name: string not null
table_type: string not null
----
table_catalog: [["public","public","public","public","public",...,"public","public","public","public","public"]]
table_schema: [["iox","iox","iox","iox","iox",...,"iox","information_schema","information_schema","information_schema","information_schema"]]
table_name: [["earthquake","explosion","home","homeSensor","ice quake",...,"temperature","tables","views","columns","df_settings"]]
table_type: [["BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE",...,"BASE TABLE","VIEW","VIEW","VIEW","VIEW"]]
>>> client = InfluxDBClient3(host=host, token=token, database=database)
>>> client.query('show tables')
pyarrow.Table
table_catalog: string not null
table_schema: string not null
table_name: string not null
table_type: string not null
----
table_catalog: [["public","public","public","public","public",...,"public","public","public","public","public"]]
table_schema: [["iox","iox","iox","iox","iox",...,"iox","information_schema","information_schema","information_schema","information_schema"]]
table_name: [["earthquake","explosion","home","homeSensor","ice quake",...,"temperature","tables","views","columns","df_settings"]]
table_type: [["BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE",...,"BASE TABLE","VIEW","VIEW","VIEW","VIEW"]]
>>> 
```

**Pull request merged**